cdn_lambda/functions/map_to_s3/map_to_s3.py

```diff
@@ -31,6 +31,12 @@ def db_client(self):
 
         return self._db_client
 
+    def uri_alias(self, uri):
+        for key, value in self.conf["uri_aliases"].items():
```

Comment:
Question: are the items guaranteed to be processed here in the same order they were listed in the JSON file?  Because the code only seems to work correctly if that's the case.  (For instance /content/origin/rpm/foobar only works if /content/origin/ is processed and then /origin/rpm/ is processed.)

At the very least I would recommend adding a comment here pointing out that order matters here, as it's not at all obvious.  Maybe even going further and making the config file use a list instead of a dict.

Reply:
It seems that, in Python3 at least, the order in the JSON file is maintained. I considered sorting the dict to ensure /content/ was always processed first but having the dict alphabetized may cause issues for other alias sets.

I'm not sure how the dict could be converted into a list or how that might help.

Reply:
> I'm not sure how the dict could be converted into a list or how that might help.

We control both the config file format and the code, so you wouldn't convert it to a list, you would just make the config file use a list in the first place, e.g.

```
  "uri_aliases": [
    ["/content/origin/", "/origin/"],
    ["/origin/rpm/", "/origin/rpms/"]
  ]
```

It would help because lists always retain order, unlike dicts which generally have an undefined order.

Do we deploy on at least Python 3.7?  Per comments at https://docs.python.org/3/library/json.html (search for "preserve input and output order"), that seems to be the first version at which this behavior can be relied on, that was the first Python version where dicts became ordered by default.  tox.ini is only testing with Python 3.6.

Reply:
Gotcha. I didn't know list order was always maintained. In that case, I'll switch to a 2D-list, as you suggest.